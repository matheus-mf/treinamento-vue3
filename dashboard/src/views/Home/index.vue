<template>
  <CustomHeader
    @create-account="handleAccountCreate"
    @login="handleLogin"
  />
  <contact/>
  <footer class="flex justify-center py-10 bg-brand-gray">
    <p class="font-medium text-center text-gray-800">feedback &copysr; 2021</p>
  </footer>
</template>
<script>
import { onMounted } from 'vue'
import { useRouter } from 'vue-router'
import CustomHeader from './CustomHeader'
import Contact from './Contact'
import useModal from '../../hooks/useModal'

export default {
  components: { Contact, CustomHeader },
  setup () {
    const router = useRouter()
    const { open } = useModal()

    onMounted(() => {
      const token = window.localStorage.getItem('token')
      if (token) {
        router.push({ name: 'Feedbacks' })
      }
    })

    function handleLogin () {
      open({
        component: 'ModalLogin'
      })
    }

    function handleAccountCreate () {
      open({
        component: 'ModalCreateAccount'
      })
    }

    return {
      handleAccountCreate,
      handleLogin
    }
  }
}
</script>
