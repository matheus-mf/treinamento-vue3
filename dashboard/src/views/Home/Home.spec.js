import { createRouter, createWebHistory } from 'vue-router'
import { shallowMount } from '@vue/test-utils'

import { routes } from '../../router'

import Home from '.'

const router = createRouter({
  history: createWebHistory('/'),
  routes
})

describe('<Home />', () => {
  it('should render home correctly', async () => {
    router.push('/')
    await router.isReady()
    const wrapper = shallowMount(Home, {
      global: {
        plugins: [router]
      }
    })

    expect(wrapper.html()).toMatchSnapshot()
  })
})
