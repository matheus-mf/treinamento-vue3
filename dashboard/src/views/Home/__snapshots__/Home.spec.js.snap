// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> should render home correctly 1`] = `
<custom-header-stub></custom-header-stub>
<contact-stub></contact-stub>
<footer class="flex justify-center py-10 bg-brand-gray">
  <p class="font-medium text-center text-gray-800">feedback â„— 2021</p>
</footer>
`;
